### ISSUE TITLE
Cannot Remove Attached Controlled Document from Form

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/HPI#1120]
When using Alfresco, it seems that I cannot remove an attached controlled document from a wizard form.  Here's what I'm seeing on edge2:

![image](https://cloud.githubusercontent.com/assets/1377228/16050929/2e294012-3223-11e6-91fd-17b4f3a9beec.png)

![image](https://cloud.githubusercontent.com/assets/1377228/16050938/376eb47c-3223-11e6-921f-f8e141880319.png)

There's no stack trace in OC logs.  Among the condition validation one `WARN` message comes through:

```
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair hasRelatePermission/true
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-relate-conditionevaluator-CONDITIONID/true
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/condition pair condition-relate-conditionevaluator-CONDITIONID/condition-relate-conditionevaluator
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-relate-conditionevaluator-CONDITIONID-MESSAGE/null
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair wfInstanceIdsForContent-workspace://SpacesStore/b7005901-6c92-4893-956e-edf4defb342f/[Ljava.lang.String;@29f8d355
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-isnotinworkflow-conditionevaluator-CONDITIONID/false
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/condition pair condition-isnotinworkflow-conditionevaluator-CONDITIONID/condition-isnotinworkflow-conditionevaluator
10:45:19,950 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-isnotinworkflow-conditionevaluator-CONDITIONID-MESSAGE/null
10:45:19,981 WARN  [com.tsgrp.opencontent.alfresco.extendedcontent.AlfrescoEmbExtendedContentImpl] [ajp-apr-8009-exec-4] Found no versions on node workspace://SpacesStore/b7005901-6c92-4893-956e-edf4defb342f with version later than 0.0.
10:45:19,981 DEBUG [com.tsgrp.opencontent.core.action.condition.evaluator.AbstractConditionEvaluator] [ajp-apr-8009-exec-4] Current version ID is workspace://SpacesStore/b7005901-6c92-4893-956e-edf4defb342f compared against doc workspace://SpacesStore/b7005901-6c92-4893-956e-edf4defb342f
10:45:19,981 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-isobjectcurrent-conditionevaluator-CONDITIONID/true
10:45:19,981 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/condition pair condition-isobjectcurrent-conditionevaluator-CONDITIONID/condition-isobjectcurrent-conditionevaluator
10:45:19,981 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-isobjectcurrent-conditionevaluator-CONDITIONID-MESSAGE/null
10:45:19,997 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair workspace://SpacesStore/b7005901-6c92-4893-956e-edf4defb342fCoreContext.getLifecycle().getCurrentState(ticket, nodeId)/{http://www.activewizard.com/model/content/1.0}formDraft
10:45:19,997 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-iswizardformindraftstate-conditionevaluator-CONDITIONID/true
10:45:19,997 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/condition pair condition-iswizardformindraftstate-conditionevaluator-CONDITIONID/condition-iswizardformindraftstate-conditionevaluator
10:45:19,997 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-iswizardformindraftstate-conditionevaluator-CONDITIONID-MESSAGE/null
10:45:19,997 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-iswizardformincancelledstate-conditionevaluator-CONDITIONID/false
10:45:19,997 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/condition pair condition-iswizardformincancelledstate-conditionevaluator-CONDITIONID/condition-iswizardformincancelledstate-conditionevaluator
10:45:19,997 DEBUG [com.tsgrp.opencontent.core.action.EvaluationCache] [ajp-apr-8009-exec-4] caching key/val pair condition-iswizardformincancelledstate-conditionevaluator-CONDITIONID-MESSAGE/null

```

The noderef it mentions in that message is the form ID, not the document ID.

**Comment by gsteimer** [comment]
Removing the Alfresco label.  I just tried this on Documentum, and the same error displays in the HPI interface.  Nothing is writing to the OC logs that looks out of the ordinary other than:

```
2016-06-14 11:46:50,969	41654477	TP-Processor6	DEBUG	AbstractConditionEvaluator	Current version ID is 09000002800133c3 compared against doc 09000002800133c3
```

**Comment by gsteimer** [comment]
This was resolved with the fix to #1112
